Client/Frontend/Settings/BraveRewardsSettingsViewController.swift

```diff
@@ -47,25 +52,53 @@ class BraveRewardsSettingsViewController: TableViewController {
-                    ]
-                ), at: 1)
-            })
+        if let legacyWallet = legacyWallet {
+            if Preferences.Rewards.transferDrainID.value == nil {
+                if !legacyWallet.isLedgerTransferExpired {
+                    legacyWallet.transferrableAmount({ [weak self] total in
+                        guard let self = self, total > 0 else { return }
+                        self.dataSource.sections.insert(.init(
+                            header: .title(Strings.Rewards.walletTransferTitle),
+                            rows: [
+                                Row(text: Strings.Rewards.legacyWalletTransfer,
+                                    detailText: Preferences.Rewards.lastTransferStatus.value.map(DrainStatus.init)??.displayString,
+                                    selection: { [unowned self] in
+                                        guard let legacyWallet = self.legacyWallet else { return }
+                                        let controller = WalletTransferViewController(legacyWallet: legacyWallet)
+                                        controller.learnMoreHandler = { [weak self] in
+                                            self?.walletTransferLearnMoreTapped?()
+                                        }
+                                        let container = UINavigationController(rootViewController: controller)
+                                        container.modalPresentationStyle = .formSheet
+                                        self.present(container, animated: true)
+                                    }, image: UIImage(imageLiteralResourceName: "rewards-qr-code").template)
+                            ]
+                        ), at: 1)
+                    })
+                }
+            } else {
+                // Check to see if the transfer already completed and was acknowledged by the user,
+                // so we dont show the row for wallet transfer anymore
+                if !Preferences.Rewards.transferCompletionAcknowledged.value {
+                    legacyWallet.updateDrainStatus { status in
+                        if let status = status {
+                            self.dataSource.sections.insert(.init(
+                                header: .title(Strings.Rewards.walletTransferTitle),
+                                rows: [
+                                    Row(text: Strings.Rewards.legacyWalletTransfer,
+                                        detailText: status.displayString,
+                                        selection: { [unowned self] in
+                                            let controller = WalletTransferCompleteViewController(status: status)
+                                            let container = UINavigationController(rootViewController: controller)
+                                            container.modalPresentationStyle = .formSheet
```

Comment:
Is this ok for both ipads and iphones?

Reply:
Yeah on iPhone it presents as a regular sheet, makes me wonder if we should just remove all the other checks we do like this haha.

One thing I just realized though is I have to set `isModalInPresentation` to true if the status is completed (since they need to manually tap Done to acknowledge)